go/vt/vtgate/evalengine/eval.go

```diff
@@ -130,6 +130,9 @@ func evalToSQLValueWithType(e eval, resultType sqltypes.Type) sqltypes.Value {
 			return sqltypes.MakeTrusted(resultType, e.dec.FormatMySQL(e.length))
 		}
 	default:
+		if e == nil {
+			return sqltypes.MakeTrusted(resultType, nil)
+		}
```

Comment:
it was panicking on `e.ToRawBytes()` since `e` is `nil`

Reply:
But is this new returned value the right one? Like what if this is an integer type, a `nil` value for the bytes is afaik not valid? Should it be for example `0` in that case? 

Or should this here return `sqltypes.NULL` instead of this default type? Does this potentially point at a problem in the caller instead where `NULL` is maybe not handled correctly?

Reply:
This was fixed in https://github.com/vitessio/vitess/pull/14906/commits/26aef4ab185d90de70e84d84032d5c8ddb971097